### PR TITLE
Change pydantic to be compatible with typing-extension==4.6.1

### PR DESCRIPTION
**Pull request by hramezani**
port https://github.com/pydantic/pydantic/commit/c4d3dee621884670eabbeb9b57b973508193c664

Selected Reviewer: @dmontagu

### CODE REVIEW

**Comment by hramezani** [comment]
please review

**Review by adriangb — APPROVED** [review]
Do we need to backport the `pydantic/v1` subdirectory?

File: pydantic/_internal/_typing_extra.py [hunk 1]
```diff
@@ -93,18 +96,15 @@ def is_none_type(type_: Any) -> bool:
 else:
 
     def is_none_type(type_: Any) -> bool:
-        for none_type in NONE_TYPES:
-            if type_ is none_type:
-                return True
-        return False
+        return type_ in NONE_TYPES
```
[the remark this answers is not in the document]
Comment (hramezani) [reply pydantic/_internal/_typing_extra.py]: Done!
I copied the changes from your PR

**Comment by hramezani** [comment]
> Do we need to backport the `pydantic/v1` subdirectory?

I've updated the `v1` by the lastest `1.10.8`.

File: pydantic/_internal/_typing_extra.py [hunk 2]
```diff
@@ -468,3 +444,7 @@ def is_dataclass(_cls: type[Any]) -> TypeGuard[type[StandardDataclass]]:
     # The dataclasses.is_dataclass function doesn't seem to provide TypeGuard functionality,
     # so I created this convenience function
     return dataclasses.is_dataclass(_cls)
+
+
+def origin_is_type_alias_type(origin: Any) -> TypeGuard[TypeAliasType]:
+    return isinstance(origin, TypeAliasType)
```
Comment (adriangb) [inline pydantic/_internal/_typing_extra.py:450]:
```suggestion

```

**Review by adriangb — COMMENTED** [review]
Looks like you may have copied over a bit too much

**Comment by adriangb** [comment]
Looks good! Please merge when you feel this is ready.

**Comment by hramezani** [comment]
Ok, it's ready @adriangb

**Pull request merged**